### PR TITLE
`azurerm_iothub` - Support for `public_network_access_enabled`

### LINKED ISSUE #8508: Azure IotHub provider - Allow public access to be Disable property not supported
### Community Note

Azure IotHub provider - Allow public access to be Disable property not supported

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by mbfrahry**
Fixes #8508 

image

### CODE REVIEW

File: azurerm/internal/services/iothub/iothub_resource.go [hunk 1]
```diff
@@ -403,6 +403,11 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"public_network_access": {
+				Type:     schema.TypeBool,
+				Optional: true,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/iothub/iothub_resource.go:408]: does this need to default to enabled? it'd be good to confirm with the portal's behaviour

File: azurerm/internal/services/iothub/iothub_resource.go [hunk 2]
```diff
@@ -621,6 +635,13 @@ func resourceArmIotHubRead(d *schema.ResourceData, meta interface{}) error {
 		if err := d.Set("file_upload", fileUpload); err != nil {
 			return fmt.Errorf("Error setting `file_upload` in IoTHub %q: %+v", name, err)
 		}
+
+		if enabled := properties.PublicNetworkAccess; enabled != "" {
+			d.Set("public_network_access", false)
+			if enabled == devices.Enabled {
+				d.Set("public_network_access", true)
+			}
+		}
```
Comment (tombuildsstuff) [inline azurerm/internal/services/iothub/iothub_resource.go:644]: couldn't we could make this:

```suggestion
		publicNetworkAccessEnabled := properties.PublicNetworkAccess == devices.Enabled
		d.Set("public_network_access", publicNetworkAccessEnabled)
```

**Review by tombuildsstuff — COMMENTED** [review]
a couple of questions but this otherwise LGTM 👍

File: azurerm/internal/services/iothub/iothub_resource.go (hunk 1 above)
Comment (mbfrahry) [reply azurerm/internal/services/iothub/iothub_resource.go:408]: If you don't specify it when creating an iothub, it doesn't get returned by the api so I don't think we can default it without breaking people.

File: azurerm/internal/services/iothub/iothub_resource.go (hunk 2 above)
Comment (mbfrahry) [reply azurerm/internal/services/iothub/iothub_resource.go:644]: Sure could

File: azurerm/internal/services/iothub/iothub_resource.go (hunk 1 above)
Comment (mbfrahry) [reply azurerm/internal/services/iothub/iothub_resource.go:408]: You're required to specify it when creating it in the portal. I believe this functionality was added recently. I imagine it's on by default but the api doesn't return it if a user didn't specify it when creating. 

image

**Comment by mbfrahry** [comment]
This is ready for another review @tombuildsstuff

File: azurerm/internal/services/iothub/iothub_resource.go [hunk 3]
```diff
@@ -403,6 +403,11 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"public_network_access": {
```
Comment (katbyte) [inline azurerm/internal/services/iothub/iothub_resource.go:406]: Should this be `public_network_access_enabled` ?

**Review by katbyte — APPROVED** [review]
Aside from one comment about the property name LGTM ! 🚀

**Pull request merged**